activerecord/test/actioncontroller/query_comments_test.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+require "cases/helper"
+require "action_controller"
+require "active_record/railties/query_comments"
+require "models/dashboard"
+
+class DashboardController < ActionController::Base
+  def index
+    @dashboard = Dashboard.first
+    render body: nil
+  end
+end
+
+class DashboardApiController < ActionController::API
+  def index
+    render json: Dashboard.all
+  end
+end
+
+ActionController::Base.include(ActiveRecord::Railties::QueryComments::ActionController)
+ActionController::API.include(ActiveRecord::Railties::QueryComments::ActionController)
+ActiveRecord::ConnectionAdapters::AbstractAdapter::QueryCommentContext.components = [:application, :controller, :action]
```

Comment:
Do these persist? If `QueryCommentContext.components` does it should go in a setup/teardown block like this:

```ruby
def setup
  @previous_components = ActiveRecord::ConnectionAdapters::AbstractAdapter::QueryCommentContext.components
  ActiveRecord::ConnectionAdapters::AbstractAdapter::QueryCommentContext.components = [:application, :controller, :action]
end

def teardown
  ActiveRecord::ConnectionAdapters::AbstractAdapter::QueryCommentContext.components = @previous_components
end
```

Reply:
Hmm good point. These are essentially reproducing what happens [in the railtie](https://github.com/keeran/rails/pull/1/files#diff-41177c56a83ce1447bfba614c4b7f29f3deef57f2312a9e0591763c489e76099R92-R94), but agree other tests could change the setup / expect a different setup.